packages/admin/admin-icons/generate-icons.ts

```diff
@@ -63,12 +80,19 @@ const getFormattedText = async (text: string) => {
     return results[0].output;
 };
 
-const writeComponent = async (componentName: string, pathData: string) => {
+const writeComponent = async (icon: Icon, pathData: string) => {
     const component = await getFormattedText(`
         import { SvgIcon, SvgIconProps } from "@mui/material";
         import * as React from "react";
         
-        export default function ${componentName}(props: SvgIconProps): JSX.Element {
+        ${
+            icon.deprecated
+                ? `/**
+                    * @deprecated Do not use.
```

Comment:
I would use a more descriptive deprecation message. IDEs will already tell you that it shouldn't be used.

Reply:
What would be more descriptive, but still generic? `The '<name>' icon will be removed in a future major release.`? Do I need to include the name (as it's already the component name)?

Reply:
That's up to you. Either `This icon will be removed in a future major release.` or `The '<name>' icon will be removed in a future major release.` are completely fine :)